### ISSUE TITLE
[Bug] optimize_acqf_mixed changes points in fixed_feature_list

### ISSUE THREAD

**Issue by AVHopp** [pytorch/botorch#1943]
# 🐛 Bug

When fixing features using the `fixed_features_list` keyword in `optimize_acqf_mixed`, the candidates returned by the function do not have the exact values that are specified. The issue also arises when another default data type is chosen (e.g. when using `torch.set_default_dtype(torch.float32)`).

## To reproduce

```python
import torch
from botorch.models import SingleTaskGP
from botorch.acquisition import UpperConfidenceBound
from botorch.optim import optimize_acqf_mixed

torch.set_default_dtype(torch.float64)
torch.set_printoptions(precision=16)

train_X = torch.rand(10, 2)
train_Y = 1 - torch.norm(train_X - 0.5, dim=-1, keepdim=True)

gp = SingleTaskGP(train_X, train_Y)

UCB = UpperConfidenceBound(gp, beta=0.1)

fixed_features_list = [{0: 1e-1}]

bounds = torch.stack([torch.zeros(2), torch.ones(2)])
candidate, acq_value = optimize_acqf_mixed(
    UCB,
    bounds=bounds,
    q=1,
    num_restarts=5,
    raw_samples=20,
    fixed_features_list=fixed_features_list,
)
print(candidate)
```

** Stack trace/error message **
```
tensor([[0.1000000014901161, 0.4867975063328226]])
```

## Expected Behavior

I would have expected that the values that are fixed via `fixed_features_list` and the corresponding entries in `candidates` are identical. That is, the first entry of the printed tensor should be `0.1`.

## System information
- BoTorch Version 0.8.5 
- GPyTorch Version 1.10 
- PyTorch Version 2.0.1 
- Mac OS

**Comment by esantorella** [comment]
Thanks for the bug report!

> Maybe to add: `torch.tensor(0.1).to(torch.float32)` gives exactly that number, so it seems some (unintentional) float32 conversion is happening under the hood?

Yep, this is exactly what was happening. I just put up a PR to address this.

By the way, the bounds in this example are also float32. That's not contributing to this issue, but they'd need to be explicitly constructed as `float64` if that level of precision is needed.

**Comment by Balandat** [comment]
Fixed #1944
